backend/src/main/java/fk/prof/backend/http/BackendHttpVerticle.java

```diff
@@ -0,0 +1,177 @@
+    vertx.createHttpServer(HttpHelper.getHttpServerOptions(backendHttpServerConfig))
+        .requestHandler(router::accept)
+        .listen(backendHttpServerConfig.getInteger("port"), http -> completeStartup(http, fut));
+  }
+
+  private Router setupRouting() {
+    Router router = Router.router(vertx);
+    router.route().handler(LoggerHandler.create());
+
+    router.post(ApiPathConstants.AGGREGATOR_POST_PROFILE).handler(this::handlePostProfile);
+
+    router.post(ApiPathConstants.BACKEND_PUT_ASSOCIATION)
+        .handler(BodyHandler.create().setBodyLimit(1024 * 10));
+    router.post(ApiPathConstants.BACKEND_PUT_ASSOCIATION)
+        .handler(this::handlePutAssociation);
+
+    return router;
+  }
+
+  @Override
+  public void stop() throws Exception {
+    if(loadReportTimer != null) {
+      vertx.cancelTimer(loadReportTimer);
+    }
+  }
+
+  private void completeStartup(AsyncResult<HttpServer> http, Future<Void> fut) {
+    if (http.succeeded()) {
+      fut.complete();
+    } else {
+      fut.fail(http.cause());
+    }
+  }
+
+  private void handlePostProfile(RoutingContext context) {
+    CompositeByteBufInputStream inputStream = new CompositeByteBufInputStream();
+    RecordedProfileProcessor profileProcessor = new RecordedProfileProcessor(
+        profileWorkService,
+        new SharedMapBasedSingleProcessingOfProfileGate(workIdsInPipeline),
+        config().getJsonObject("parser").getInteger("recordingheader.max.bytes", 1024),
+        config().getJsonObject("parser").getInteger("parser.wse.max.bytes", 1024 * 1024));
+
+    RecordedProfileRequestHandler requestHandler = new RecordedProfileRequestHandler(context, inputStream, profileProcessor);
+    context.request()
+        .handler(requestHandler)
+        .endHandler(v -> {
+          try {
+            if (!context.response().ended()) {
+              //Can safely attempt to close the profile processor here because endHandler is called once the entire body has been read
+              //and example in vertx docs also indicates that this handler will execute once all chunk handlers have completed execution
+              //http://vertx.io/docs/vertx-core/java/#_handling_requests
+              inputStream.close();
+              profileProcessor.close();
+              context.response().end();
+            }
+          } catch (Exception ex) {
+            HttpFailure httpFailure = HttpFailure.failure(ex);
+            HttpHelper.handleFailure(context, httpFailure);
+          }
+        });
+  }
+
+  private void handlePutAssociation(RoutingContext context) {
+    String leaderIPAddress = getLeaderAddressOrAbortRequest(context.response());
+    if (leaderIPAddress != null) {
+      try {
+        //Deserialize to proto message to catch payload related errors early
+        Recorder.ProcessGroup processGroup = Recorder.ProcessGroup.parseFrom(context.getBody().getBytes());
+        makeRequestGetAssociation(leaderIPAddress, processGroup).setHandler(ar -> {
```

Comment:
Yes, I should have added a TODO here. Backend side of things(association store) is not implemented yet. Once that is done in next PR, I should add that check in this API request